packages/cli/src/__tests__/commands/Generate.test.ts

```diff
@@ -191,6 +192,55 @@ describe('--schema from parent directory', () => {
   })
 })
 
+describe('--binary-target', () => {
+  const getQueryEngineFileName = (platform: Platform) =>
+    match(getClientEngineType())
+      .with(ClientEngineType.Library, () => getNodeAPIName(platform, 'fs'))
+      .with(
+        ClientEngineType.Binary,
+        () =>
+          // TODO: this is duplicated throughout the codebase, factor it out
+          `query-engine-${platform}${platform === 'windows' ? '.exe' : ''}`,
+      )
+      .exhaustive()
+
+  it('must not implicitly include "native"', async () => {
```

Comment:
This makes the CLI option a non-breaking solution to https://github.com/prisma/prisma/issues/12691, suggested by @dejanvasic85 in the comments.

Reply:
Actually after checking I don't even think what's written in https://github.com/prisma/prisma/issues/12691 is true (at least anymore): we don't implicitly add `native` to `binaryTargets` in schema either but print a warning instead. `native` is only added implicitly when no binary targets are specified in the schema.